syntax = "proto3";

package endpoints.word_service;

enum DatasetType {
    OED = 0;
    UD_FILTERED = 1;
    UD_UNFILTERED = 2;
}

service WordService {
  rpc DefineWord(DefineWordRequest) returns (DefineWordResponse) {}
  rpc WordFromDefinition(WordFromDefinitionRequest) returns (WordFromDefinitionResponse) {}
  rpc GenerateWord(GenerateWordRequest) returns (GenerateWordResponse) {}
}

message WordDefinition {
  string word = 1;
  string definition = 2; 
  string pos = 3;
  repeated string examples = 4;
  repeated string syllables = 5;
  bool probablyExists = 6;
  DatasetType dataset = 7;
}

message DefineWordRequest {
  string word = 1;
  DatasetType dataset = 2;
}

message DefineWordResponse {
  WordDefinition word = 1;
}

message WordFromDefinitionRequest {
  string definition = 1;
}
message WordFromDefinitionResponse {
  WordDefinition word = 1;
}

message GenerateWordRequest {}
message GenerateWordResponse {
  WordDefinition word = 1;
}
